src/VisualStudio/Core/Def/Packaging/PackageInstallerService.ProjectState.cs

```diff
@@ -2,22 +2,37 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using Roslyn.Utilities;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
 
 namespace Microsoft.VisualStudio.LanguageServices.Packaging
 {
     internal partial class PackageInstallerService
     {
         private struct ProjectState
         {
+            public static readonly ProjectState Disabled = new ProjectState(isEnabled: false, ImmutableDictionary<string, string>.Empty);
+
             public readonly bool IsEnabled;
-            public readonly MultiDictionary<string, string> InstalledPackageToVersion;
 
-            public ProjectState(bool isEnabled, MultiDictionary<string, string> installedPackageToVersion)
+            private readonly ImmutableDictionary<string, string> InstalledPackageToVersion;
+
+            private ProjectState(bool isEnabled, ImmutableDictionary<string, string> installedPackageToVersion)
             {
                 IsEnabled = isEnabled;
                 InstalledPackageToVersion = installedPackageToVersion;
             }
+
+            public ProjectState(ImmutableDictionary<string, string> installedPackageToVersion)
+                : this(isEnabled: true, installedPackageToVersion)
+            {
+            }
+
+            public bool IsInstalled(string package)
+                => IsEnabled && InstalledPackageToVersion.ContainsKey(package);
+
+            public bool TryGetInstalledVersion(string packageName, [NotNullWhen(true)] out string version)
```

Comment:
Is this file missing a `#nullable enable` since you're otherwise annotating it?

Reply:
done.